pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -208,10 +211,103 @@ func (n *nodeHealthMap) set(name string, data *nodeHealthData) {
 	n.nodeHealths[name] = data
 }
 
+type podUpdateItem struct {
+	namespace string
+	name      string
+}
+
+type evictionStatus int
+
+const (
+	unmarked = iota
+	toBeEvicted
+	evicted
+)
+
+type nodeEviction struct {
+	status evictionStatus
+	pods   map[types.UID]*v1.Pod
+}
+
+type nodeEvictionMap struct {
+	lock          sync.Mutex
+	nodeEvictions map[string]*nodeEviction
+}
+
+func newNodeEvictionMap() *nodeEvictionMap {
+	return &nodeEvictionMap{
+		nodeEvictions: make(map[string]*nodeEviction),
+	}
+}
+
+func (n *nodeEvictionMap) addNodeEviction(nodeName string) {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+	if _, exists := n.nodeEvictions[nodeName]; !exists {
```

Comment:
given where this is called (and if we name it registerNode), we probably don't need to check it - we can do that unconditionally